src/SDKs/Batch/DataPlane/Azure.Batch.Unit.Tests/BatchSharedKeyCredentialUnitTests.cs

```diff
@@ -0,0 +1,90 @@
+
+namespace Azure.Batch.Unit.Tests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Net.Http;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Batch.Protocol;
+    using Microsoft.Net.Http.Server;
+    using Xunit;
+
+    public class BatchSharedKeyCredentialUnitTests
+    {
+        private const string url = "http://localhost:2055";
+
+        [Theory]
+        [MemberData(nameof(HttpMethods))]
+        public async Task TestSharedKeySigning_ContentLength(HttpMethod httpMethod)
+        {
+            BatchSharedKeyCredential creds = new BatchSharedKeyCredential(ClientUnitTestCommon.DummyAccountName, ClientUnitTestCommon.DummyAccountKey);
+
+            HttpRequestMessage message = new HttpRequestMessage(httpMethod, url);
+            message.Headers.Add("client-request-id", Guid.NewGuid().ToString());
+
+            await creds.ProcessHttpRequestAsync(message, CancellationToken.None);
+            Assert.NotNull(message.Headers.Authorization);
+
+            void AssertLambda(RequestContext ctx)
+            {
+                if (httpMethod == HttpMethod.Head || httpMethod == HttpMethod.Get)
+                {
+                    Assert.DoesNotContain(ctx.Request.Headers.Keys, str => str == "Content-Length");
+                }
+                else if (httpMethod == HttpMethod.Delete)
+                {
+#if !FullNetFx
+                    Assert.DoesNotContain(ctx.Request.Headers.Keys, str => str == "Content-Length");
+#else
+                    Assert.Contains(ctx.Request.Headers.Keys, str => str == "Content-Length");
+#endif
+                }
+                else if (httpMethod == HttpMethod.Post || httpMethod == HttpMethod.Put)
+                {
+                    Assert.Contains(ctx.Request.Headers.Keys, str => str == "Content-Length");
+                }
+                else
+                {
+                    throw new ArgumentException($"Unexpected HTTP request type: {httpMethod}");
+                }
+            }
+
+            var settings = new WebListenerSettings()
+            {
+                Authentication = { Schemes = AuthenticationSchemes.None },
+                UrlPrefixes = { url }
+            };
+            using (WebListener listener = new WebListener(settings))
+            {
+                listener.Start();
+                Task listenTask = AcceptAndAssertAsync(listener, AssertLambda);
+
+                HttpClient client = new HttpClient();
+                await client.SendAsync(message);
+
+                await listenTask;
+            }
+        }
+
+        private static IEnumerable<object[]> HttpMethods()
+        {
+            yield return new[] { HttpMethod.Delete };
+            yield return new[] { HttpMethod.Post };
+            yield return new[] { HttpMethod.Get };
+            yield return new[] { HttpMethod.Head };
+            yield return new[] { HttpMethod.Put };
```

Comment:
Daryl brought this up too... even though we don't use `Options` anywhere internally, I guess I'll add it.